torch/distributions/utils.py

```diff
@@ -85,6 +86,45 @@ def probs_to_logits(probs, is_binary=False):
     return torch.log(ps_clamped)
 
 
+def continued_fraction(a, b, value, niters=2000, tol=3.0e-11):
+    r"""
+    Evaluates the continued fraction form of the Incomplete Beta Function,
+    required for Beta CDF.
+
+    Adapted from Numerical Recipes in C, Section 6.4
+    (http://www.aip.de/groups/soe/local/numres/bookcpdf/c6-4.pdf)
+
+    Args:
+        a (Tensor): concentration1 of Beta distribution
+        b (Tensor): concentration0 of Beta distribution
+        value (Tensor): value provided of Beta CDF function
+        niters (int): Maximum number of iterations to find convergence.
+    """
+    bm = az = am = 1.0
+    qap = a + 1.0
+    qam = a - 1.0
+    bz = 1.0 - (a + b) * value / qap
+
+    for i in range(niters + 1):
+        em = float(i + 1)
+        tem = em + em
+        d = em * (b - em) * value / ((qam + tem) * (a + tem))
+        ap = az + d * am
+        bp = bz + d * bm
+        d = -(a + em) * ((a + b) + em) * value / ((qap + tem) * (a + tem))
+        app = ap + d * az
+        bpp = bp + d * bz
+        a_old = az
+        am = ap / bpp
+        bm = bp / bpp
+        az = app / bpp
+        bz = 1.0
+        if torch.where(abs(az - a_old) < (tol * abs(az))):
```

Comment:
you might want to check the docs for torch.where https://pytorch.org/docs/stable/torch.html#torch.where